### PR TITLE
completed crosskick mvp

### PR DESCRIPTION
**Pull request by Jason27chan**

### CODE REVIEW

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+import tactics
+import evaluation
+import main
+import enum
+import behavior
+
+class CrossKick(play.Play):
+
+    class State(enum.Enum):
+    	collecting = 0;
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Can you add a description for all the states. With all the different transitions and states the readability is a little tough.

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    	transitioningDL = 9;
+
+
+    def __init__(self):
+        super().__init__(continuous=False)
+
+        for s in CrossKick.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        ready_to_finish = (main.ball().pos.y >= constants.Field.Length * 0.8)
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Is this being continuously updated?

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        isLeft = (main.ball().pos.x < 0) & (main.ball().pos.y >= constants.Field.Length * 0.45)
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Same question for all of these values

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+        capture = skills.capture.Capture()
+        self.add_subbehavior(capture, 'capture', required=True) 
+
+    def on_exit_collecting(self):
+        #self.captureRobot = self.subbehavior_with_name('capture')
+        self.remove_all_subbehaviors()
+
+    
+    def on_enter_transitioning(self):
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Should you rename this to on_enter transitioning_right? Just to give more description of what you're doing?

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+        #capture = skills.capture.Capture()
+
+        self.add_subbehavior(pass_to_flank, 'pass to flank', required=True)
+        #self.add_subbehavior(move_flank, 'move to flank', required=True)
+        #self.add_subbehavior(capture, 'capture', required=True) 
+
+    def on_exit_transitioningLeft(self):
+        self.remove_all_subbehaviors()
+   
+    def on_enter_transitioningDeep(self):
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Same question should this be on_enter_transitioning_deep_right?

File: soccer/gameplay/plays/offense/cross_kick.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+
+    def on_exit_transitioningDL(self):
+        self.remove_all_subbehaviors()
+    
+    def on_enter_crossing(self):
+        self.remove_all_subbehaviors()
+        
+        #capture = skills.capture.Capture()
+        #point_near_box = robocup.Point(-.4,9*.83)
+        point_in_box = robocup.Point(constants.Field.Width * -0.1, constants.Field.Length * .875)
```
Comment (tjones320) [inline soccer/gameplay/plays/offense/cross_kick.py]: Are we ok to having a hard coded points in general? What do we do if an enemy robot is in that position? Should we use any other kick location selector?

File: soccer/gameplay/plays/training/basic_offense.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import robocup
+import constants
+import play
+import skills
+import tactics
+import evaluation
+import main
+
+class BasicOffense(play.Play):
```
Comment (tjones320) [inline soccer/gameplay/plays/training/basic_offense.py]: What's the purpose of these changes?

**Comment by JNeiger** [comment]
What's this play supposed to do? I think I might have missed this specific project

**Comment by kylestach** [comment]
@hdemusg what's the status on this? Is it still in development?

**Comment by hdemusg** [comment]
I will probably make the changes that were suggested and then call it a day
on this play.

**Comment by Jason27chan** [comment]
Needs some major reworking in a code sense and also in a situational analysis sense. Also havent seen @hdemusg in a while. Closing for now
